lnrpc/routerrpc/router.proto

```diff
@@ -1063,6 +1063,14 @@ message ForwardHtlcInterceptResponse {
     // Any custom records that should be set on the p2p wire message message of
     // the resumed HTLC. This field is ignored if the action is not
     // RESUME_MODIFIED.
+    //
+    // If this map is populated, it will overwrite any of the wire custom
+    // records set by LND. It is the caller's responsibility to copy any desired
+    // records across. If the map is empty, no action will be taken and existing
+    // custom records will be propagated.
+    //
+    // The API does not currently support deleting custom records, unless they
+    // are overwritten by a new set.
```

Comment:
This seems to follow by virtue of it being a replacement API. Not sure we need this second part.

Reply:
I think it's okay to leave this in. It also kind of implies that if you want to delete (e.g. _NOT_ forward) all custom records, you can't do that currently (because you'd want to replace with an empty map which then causes the logic to do nothing because the map is empty)...

Reply:
> This seems to follow by virtue of it being a replacement API

This is kinda the crux - it's sometimes a replacement API and sometimes a merge API. Could remove " unless they are overwritten by a new set" if it's blocking?

Reply:
Oof. No this comment is fine as is. However, it seems like we should have not mixed approaches (replace vs merge), though that critique is both water under the bridge *and* squarely out of scope here.

Reply:
We haven't shipped the new API in a released version yet. How would you propose to design it? See discussions in https://github.com/lightningnetwork/lnd/issues/9166.

Reply:
Will download discussion and opine on that issue.